src/Event/Sqs/SqsEventFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php declare(strict_types=1);
+
+namespace Bref\Event\Sqs;
+
+use Bref\Event\InvalidLambdaEvent;
+use InvalidArgumentException;
+
+final class SqsEventFactory implements SqsEventFactoryInterface
+{
+    public function createFromPayload(array $payload): SqsEvent
+    {
+        if (! isset($payload['Records'])) {
+            throw new InvalidLambdaEvent('SQS', $payload);
+        }
+
+        return new SqsEvent(array_map([$this, 'createSqsRecord'], $payload['Records']), $payload);
```

Comment:
With this "early parsing" approach, all data of the event will be parsed even if not used. In the context of such a low-level library, I'd rather avoid that and favor lazy loading.

This isn't a deal breaker though, just something I'd avoid.